## DADOS ENEM 2022

Dataset extraído de:
https://www.gov.br/inep/pt-br/acesso-a-informacao/dados-abertos/microdados/enem

In [2]:
# Importar o dataset
import pandas as pd

df = pd.read_csv(r"C:\Users\bserpellone\Desktop\Python\Dataset\ENEM 2022\DADOS\MICRODADOS_ENEM_2022.csv", encoding='ANSI', delimiter=';')

### CONHECENDO OS DADOS

In [4]:
df.shape

(3476105, 76)

In [5]:
df.columns

Index(['NU_INSCRICAO', 'NU_ANO', 'TP_FAIXA_ETARIA', 'TP_SEXO',
       'TP_ESTADO_CIVIL', 'TP_COR_RACA', 'TP_NACIONALIDADE', 'TP_ST_CONCLUSAO',
       'TP_ANO_CONCLUIU', 'TP_ESCOLA', 'TP_ENSINO', 'IN_TREINEIRO',
       'CO_MUNICIPIO_ESC', 'NO_MUNICIPIO_ESC', 'CO_UF_ESC', 'SG_UF_ESC',
       'TP_DEPENDENCIA_ADM_ESC', 'TP_LOCALIZACAO_ESC', 'TP_SIT_FUNC_ESC',
       'CO_MUNICIPIO_PROVA', 'NO_MUNICIPIO_PROVA', 'CO_UF_PROVA',
       'SG_UF_PROVA', 'TP_PRESENCA_CN', 'TP_PRESENCA_CH', 'TP_PRESENCA_LC',
       'TP_PRESENCA_MT', 'CO_PROVA_CN', 'CO_PROVA_CH', 'CO_PROVA_LC',
       'CO_PROVA_MT', 'NU_NOTA_CN', 'NU_NOTA_CH', 'NU_NOTA_LC', 'NU_NOTA_MT',
       'TX_RESPOSTAS_CN', 'TX_RESPOSTAS_CH', 'TX_RESPOSTAS_LC',
       'TX_RESPOSTAS_MT', 'TP_LINGUA', 'TX_GABARITO_CN', 'TX_GABARITO_CH',
       'TX_GABARITO_LC', 'TX_GABARITO_MT', 'TP_STATUS_REDACAO',
       'NU_NOTA_COMP1', 'NU_NOTA_COMP2', 'NU_NOTA_COMP3', 'NU_NOTA_COMP4',
       'NU_NOTA_COMP5', 'NU_NOTA_REDACAO', 'Q001', 'Q002', 'Q003', 'Q004',

#### Colunas de interesse:
TP_FAIXA_ETARIA: {
                      1: 'Menor de 17 anos', 
                      2: '17 anos',
                      3: '18 anos', 
                      4: '19 anos',
                      5: '20 anos', 
                      6: '21 anos', 
                      7: '22 anos',
                      8: '23 anos',
                      9: '24 anos', 
                      10: '25 anos', 
                      11: 'Entre 26 e 30 anos', 
                      12: 'Entre 31 e 35 anos', '13': 'Entre 36 e 40 anos', '14': 'Entre 41 e 45 anos',
                      15: 'Entre 46 e 50 anos', '16': 'Entre 51 e 55 anos', '17': 'Entre 56 e 60 anos', 
                      18: 'Entre 61 e 65 anos', '19': 'Entre 66 e 70 anos', '20': 'Maior de 70 anos'
                  }                    
TP_SEXO: M e F
TP_COR_RACA:  {
                    0: 'Não declarado', 
                    1: 'Branca',
                    2: 'Preta', 
                    3: 'Parda',
                    4: 'Amarela', 
                    5: 'Indígena ', 
                    6: 'Não dispõe da informação'
              }

TP_ANO_CONCLUIU: Ano de Conclusão do Ensino Médio
                {
                    0: 'Não informado', 
                    1: '2021', 
                    2: '2020', 
                    3: '2019', 
                    4: '2018', 
                    5: '2017', 
                    6: '2016', 
                    7: '2015', 
                    8: '2014', 
                    9: '2013', 
                    10: '2012', 
                    11: '2011', 
                    12: '2010', 
                    13: '2009', 
                    14: '2008', 
                    15: '2007', 
                    16: 'Antes de 2007'
                 }
                 
TP_ESCOLA: Tipo de escola do Ensino Médio
          {
                1: 'Não Respondeu',
                2: 'Pública',
                3: 'Privada
          }

NO_MUNICIPIO_ESC: Nome do município da escola
NO_MUNICIPIO_PROVA: Código da Unidade da Federação da aplicação da prova
SG_UF_ESC: Sigla da Unidade da Federação da escola
[TP_PRESENCA_CN', TP_PRESENCA_CH, TP_PRESENCA_LC, TP_PRESENCA_MT] = 
                                                                    {
                                                                        0: 'Faltou à prova',
                                                                        1: 'Presente na prova',
                                                                        2: 'Eliminado na prova'
                                                                    }
                                                                        
NU_NOTA_CN:	Nota da prova de Ciências da Natureza
NU_NOTA_CH:	Nota da prova de Ciências Humanas
NU_NOTA_LC:	Nota da prova de Linguagens e Códigos
NU_NOTA_MT:	Nota da prova de Matemática
TP_LINGUA: Língua Estrangeira
            {
                0: 'Inglês',
                1: 'Espanhol'
            }
NU_NOTA_REDACAO            
                                                     

Campos Q são questões socioeconômicas

In [14]:
df.head(5)

NU_INSCRICAO  NU_ANO  TP_FAIXA_ETARIA TP_SEXO  TP_ESTADO_CIVIL  \
0  210057943671    2022               14       M                2   
1  210057516120    2022               14       M                2   
2  210057280536    2022                5       F                1   
3  210055724397    2022                6       M                1   
4  210055097896    2022                4       M                0   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  TP_ESCOLA  \
0            2                 1                1                2          1   
1            1                 1                1               16          1   
2            2                 1                1                2          1   
3            3                 1                1                2          1   
4            3                 1                1                1          1   

   ...  Q016  Q017  Q018 Q019  Q020 Q021  Q022  Q023  Q024  Q025  
0  ...     B     A     A    A     A    A     A     A     A     A  
1  ...     E     E     B    E     B    B     E     B     E     B  
2  ...     A     A     A    A     A    A     C     A     A     B  
3  ...     B     A     A    C     A    A     C     B     B     B  
4  ...     A     A     A    A     A    A     B     A     A     A  

[5 rows x 76 columns]

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3476105 entries, 0 to 3476104
Data columns (total 76 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   NU_INSCRICAO            int64  
 1   NU_ANO                  int64  
 2   TP_FAIXA_ETARIA         int64  
 3   TP_SEXO                 object 
 4   TP_ESTADO_CIVIL         int64  
 5   TP_COR_RACA             int64  
 6   TP_NACIONALIDADE        int64  
 7   TP_ST_CONCLUSAO         int64  
 8   TP_ANO_CONCLUIU         int64  
 9   TP_ESCOLA               int64  
 10  TP_ENSINO               float64
 11  IN_TREINEIRO            int64  
 12  CO_MUNICIPIO_ESC        float64
 13  NO_MUNICIPIO_ESC        object 
 14  CO_UF_ESC               float64
 15  SG_UF_ESC               object 
 16  TP_DEPENDENCIA_ADM_ESC  float64
 17  TP_LOCALIZACAO_ESC      float64
 18  TP_SIT_FUNC_ESC         float64
 19  CO_MUNICIPIO_PROVA      int64  
 20  NO_MUNICIPIO_PROVA      object 
 21  CO_UF_PROVA             int64  

In [23]:
# Há NaN na base?
df.columns.isna()

array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False])

In [149]:
#Como há colunas de interesse que são categóricas, vou puxar a descrição de cada categoria:
# Q006 é sobre a renda familiar:    
Q006 =  {
            'A': 'Nenhuma Renda', 
            'B': 'Até 1,2k',
            'C': 'De 1,2k a 1,8k',
            'D': 'De 1,8k a 2,4k',
            'E': 'De 2,4k a 3k',
            'F': 'De 3k a 3,6k',
            'G': 'De 3,6k a 4,8k',
            'H': 'De 4,8k a 6,1k',
            'I': 'De 6,1k a 7,3k',
            'J': 'De 7,3k a 8,5k',
            'K': 'De 8,5k a 9,7k',
            'L': 'De 9,7k a 10,9k',
            'M': 'De 10,9k a 12,1k',
            'N': 'De 12,1k a 14,5k',
            'O': 'De 14,5k a 18,2k',
            'P': 'De 18,2k a 24,2k',
            'Q': 'Acima de 24,2k'
        }

# TP_COR_RACA
Etnia = {
            0: 'Não declarado', 
            1: 'Branca', 
            2: 'Preta', 
            3: 'Parda', 
            4: 'Amarela', 
            5: 'Indígena', 
            6: 'Não dispõe da informação'         
        }
    
# Criar 2 colunas com essas descrições:
df['Renda'] = df['Q006'].map(Q006)
df['Etnia'] = df['TP_COR_RACA'].map(Etnia)

# PERGUNTAS

In [6]:
## Qual a quantidade de homens e de mulheres que se inscreveram no ENEM de 2022?
df['TP_SEXO'].value_counts()

TP_SEXO
F    2120519
M    1355586
Name: count, dtype: int64

In [7]:
## Qual a proporção de homens de de mulheres que se inscreveram no ENEM de 2022?
df['TP_SEXO'].value_counts(normalize=True)

TP_SEXO
F    0.610027
M    0.389973
Name: proportion, dtype: float64

In [168]:
# Qual a proporção das etnias dos inscritos no ENEM de 2022?
df['Etnia'].value_counts(normalize=True).round(2)

Etnia
Parda            0.43
Branca           0.41
Preta            0.12
Amarela          0.02
Não declarado    0.02
Indígena         0.01
Name: proportion, dtype: float64

# Abstinências

In [49]:
## Qual a quantidade de abstinências em pelo menos 1 dos 3 dias? (True = abstinência em pelo menos 1 dia)
((df['TP_PRESENCA_CN']!=1) | (df['TP_PRESENCA_CH']!=1) | (df['TP_PRESENCA_LC']!=1) | (df['TP_PRESENCA_MT']!=1)).value_counts().sort_index(ascending=False).round(2)

True     1131282
False    2344823
Name: count, dtype: int64

In [50]:
## Qual a proporção de abstinência em pelo menos 1 dos 3 dias? (True = abstinência em pelo menos 1 dia)
((df['TP_PRESENCA_CN']!=1) | (df['TP_PRESENCA_CH']!=1) | (df['TP_PRESENCA_LC']!=1) | (df['TP_PRESENCA_MT']!=1)).value_counts(normalize=True).sort_index(ascending=False).round(2)

True     0.33
False    0.67
Name: proportion, dtype: float64

In [150]:
## Qual a proporção dos grupos de renda familiar dos inscritos que faltaram ao menos 1 dia? (5 mais representativos somam 81%)
df['Renda'][(df['TP_PRESENCA_CN']!=1) | (df['TP_PRESENCA_CH']!=1) | (df['TP_PRESENCA_LC']!=1) | (df['TP_PRESENCA_MT']!=1)].value_counts(normalize=True).sort_values(ascending=False).round(2).head(5)

Renda
Até 1,2k          0.35
De 1,2k a 1,8k    0.19
De 1,8k a 2,4k    0.12
Nenhuma Renda     0.08
De 2,4k a 3k      0.07
Name: proportion, dtype: float64

In [152]:
## Dentre os inscritos que não permaneceram em pelo menos 1 dia, qual a proporção dos tiveram que se deslocar de cidade para fazer a prova? (True = cidade da prova diferente da cidade da escola)

# Primeiramente, vou criar um df pros casos com abstinência, onde o inscrito faltou em pelo menos 1 dia
df_abstinencia = df[(df['TP_PRESENCA_CN']!=1) | (df['TP_PRESENCA_CH']!=1) | (df['TP_PRESENCA_LC']!=1) | (df['TP_PRESENCA_MT']!=1)]

# Agora, vou localizar os inscritos que a coluna 'NO_MUNICIPIO_ESC' seja diferente de 'NO_MUNICIPIO_PROVA'
(df_abstinencia['NO_MUNICIPIO_ESC'] != df_abstinencia['NO_MUNICIPIO_PROVA']).value_counts(normalize=True).sort_index(ascending=False).round(2)

True     0.82
False    0.18
Name: proportion, dtype: float64

In [153]:
## Dentre os inscritos que não permaneceram em pelo menos 1 dia, qual a renda mensal da família? top 5 mais representativos (que somam 81%) (Q006: A = Até R$ 1.212 e Q = Acima de R$ 24.240) 
df_abstinencia['Renda'].value_counts(normalize=True).round(2).head(5)

Renda
Até 1,2k          0.35
De 1,2k a 1,8k    0.19
De 1,8k a 2,4k    0.12
Nenhuma Renda     0.08
De 2,4k a 3k      0.07
Name: proportion, dtype: float64

In [154]:
## Dentre os inscritos que não permaneceram em pelo menos um dia e que tiveram que se deslocar de cidade pra fazer a prova, qual a proporção da renda familiar? apenas do top 5 mais representativos (que somam 73%)

df.loc[((df['NO_MUNICIPIO_ESC']) != (df['NO_MUNICIPIO_PROVA'])), 'Renda'].value_counts(normalize=True).round(2).sort_values(ascending=False).head(5)

Renda
Até 1,2k          0.30
De 1,2k a 1,8k    0.17
De 1,8k a 2,4k    0.12
De 2,4k a 3k      0.08
Nenhuma Renda     0.06
Name: proportion, dtype: float64

## Conclusões sobre abstinências:
-> 1/3 dos inscritos não compareceram em pelo menos 1 dos 3 dias
-> dos que não compareceram em pelo menos 1 dos 3 dias, 82% tinham que fazer a prova em cidade diferente da escola que estudam
-> desse grupo da linha acima, 73% tinham renda familiar de até R$ 3k

# Indicadores socioeconômicos

In [155]:
# Quantidade de inscritos por faixa de renda
df['Renda'].value_counts(normalize=True).round(2)

Renda
Até 1,2k            0.29
De 1,2k a 1,8k      0.17
De 1,8k a 2,4k      0.12
De 2,4k a 3k        0.08
Nenhuma Renda       0.06
De 3,6k a 4,8k      0.06
De 4,8k a 6,1k      0.05
De 3k a 3,6k        0.05
De 6,1k a 7,3k      0.02
De 7,3k a 8,5k      0.02
De 9,7k a 10,9k     0.01
Acima de 24,2k      0.01
De 10,9k a 12,1k    0.01
De 14,5k a 18,2k    0.01
De 8,5k a 9,7k      0.01
De 18,2k a 24,2k    0.01
De 12,1k a 14,5k    0.01
Name: proportion, dtype: float64

In [156]:
# 5 faixas de renda mais representativas da etnia amarela (5 faixas somam 75% do total)
df[['Etnia','Renda']][df['Etnia'] == 'Amarela'].value_counts(normalize=True).round(2).head(5)

Etnia    Renda         
Amarela  Até 1,2k          0.33
         De 1,2k a 1,8k    0.17
         De 1,8k a 2,4k    0.11
         Nenhuma Renda     0.07
         De 2,4k a 3k      0.07
Name: proportion, dtype: float64

In [157]:
# 5 faixas de renda mais representativas da etnia branca (5 faixas somam 61% do total)
df[['Etnia','Renda']][df['Etnia'] == 'Branca'].value_counts(normalize=True).round(2).head(5)

Etnia   Renda         
Branca  Até 1,2k          0.18
        De 1,2k a 1,8k    0.14
        De 1,8k a 2,4k    0.12
        De 2,4k a 3k      0.09
        De 4,8k a 6,1k    0.08
Name: proportion, dtype: float64

In [158]:
# 5 faixas de renda mais representativas da etnia preta (5 faixas somam 85% do total)
df[['Etnia','Renda']][df['Etnia'] == 'Preta'].value_counts(normalize=True).round(2).head(5)

Etnia  Renda         
Preta  Até 1,2k          0.38
       De 1,2k a 1,8k    0.20
       De 1,8k a 2,4k    0.12
       Nenhuma Renda     0.08
       De 2,4k a 3k      0.07
Name: proportion, dtype: float64

In [159]:
# 5 faixas de renda mais representativas da etnia preta (5 faixas somam 83% do total)
df[['Etnia','Renda']][df['Etnia'] == 'Parda'].value_counts(normalize=True).round(2).head(5)

Etnia  Renda         
Parda  Até 1,2k          0.38
       De 1,2k a 1,8k    0.18
       De 1,8k a 2,4k    0.12
       Nenhuma Renda     0.08
       De 2,4k a 3k      0.07
Name: proportion, dtype: float64

In [160]:
# 5 faixas de renda mais representativas de indígenas (5 faixas somam 91% do total)
df[['Etnia','Renda']][df['Etnia'] == 'Indígena'].value_counts(normalize=True).round(2).head(5)

Etnia     Renda         
Indígena  Até 1,2k          0.45
          Nenhuma Renda     0.17
          De 1,2k a 1,8k    0.15
          De 1,8k a 2,4k    0.09
          De 2,4k a 3k      0.05
Name: proportion, dtype: float64

In [165]:
# Qual a proporção dos que não sabem qual a escolaridade do pai?
# Q001: Até que série seu pai, ou o homem responsável por você, estudou?
# H = Não sei

(df['Q001']=='H').value_counts(normalize=True).round(2).sort_index(ascending=False)

Q001
True     0.1
False    0.9
Name: proportion, dtype: float64

In [161]:
# Qual a proporção de quem não soube a escolaridade do pai agrupado por renda? (5 grupos mais representativos somam 76%)

df['Renda'][df['Q001']=='H'].value_counts(normalize=True).round(2).head(5)

Renda
Até 1,2k          0.41
De 1,2k a 1,8k    0.19
De 1,8k a 2,4k    0.11
Nenhuma Renda     0.09
De 2,4k a 3k      0.06
Name: proportion, dtype: float64

In [142]:
# Qual a proporção de quem não soube a escolaridade do pai agrupado por etnia?
df['Etnia'][df['Q001']=='H'].value_counts(normalize=True).round(2).head(5)

Etnia
Parda            0.48
Branca           0.31
Preta            0.15
Não declarado    0.03
Amarela          0.02
Name: proportion, dtype: float64

## Conclusões sobre questões socioeconômicas:

-> 62% dos inscritos tinham renda familiar de até R$ 3k
Analisando por etnias:
-> Brancos são a única etnia onde inscritos sem nenhuma renda não estão entre os 5 grupos de renda familiar mais representativos
-> Indígenas são os com maior proporção de sem nenhuma renda, com 17%. Nas demais etnias esse grupo não chega a 10%
-> Brancos têm a menor desigualdade de renda entre as etnias, com as 5 faixas de rendas mais representativas somando 61% do total
-> 50% dos que não sabem a escolaridade do pai têm renda familiar até R$ 1,2k
-> 76% que não sabem a escolaridade do pai têm renda familiar até R$ 3K

## Análise das notas de matemática

In [131]:
# Qual a média das notas de matemática? (Não é necessário considerar apenas quem fez a prova, o valor abaixo é o mesmo que se filtrar quem esteve presente pelo campo TP_PRESENCA_MT)
df['NU_NOTA_MT'].mean().round(2)

542.5

In [171]:
# Qual a proporção da etnia dos que fizeram a prova de matemática?
df['Etnia'][df['TP_PRESENCA_MT']==1].value_counts(normalize=True).round(2).sort_values(ascending=False)

Etnia
Branca           0.44
Parda            0.41
Preta            0.11
Amarela          0.02
Não declarado    0.02
Indígena         0.00
Name: proportion, dtype: float64

In [133]:
# Qual a proporção de quem tirou mais que 700 em matemática?
df['Matematica_maior_700'] = df['NU_NOTA_MT'] > 700
df['Matematica_maior_700'].value_counts(normalize=True).round(2).sort_index(ascending=False)

Matematica_maior_700
True     0.07
False    0.93
Name: proportion, dtype: float64

In [173]:
# Qual a proporção étnica dos que tiraram acima de 700 em matemática?
df['Etnia'][df['Matematica_maior_700']].value_counts(normalize=True).round(2).sort_values(ascending=False)

Etnia
Branca           0.67
Parda            0.25
Preta            0.04
Amarela          0.02
Não declarado    0.02
Indígena         0.00
Name: proportion, dtype: float64

In [137]:
# Qual a proporção de notas de matemática maiores que 700 conforme a renda familiar maior que R$ 4,8k?
# Q006 = Qual é a renda mensal de sua família? (Some a sua renda com a dos seus familiares.)
# Renda familiar maior que R$ 4,8k
# Q006 com respostas H em diante

df['Matematica_maior_700'][df['Q006'].isin(['H','I','J','K','L','M','N','O','P','Q'])].value_counts(normalize=True).round(2).sort_index(ascending=False)

Matematica_maior_700
True     0.21
False    0.79
Name: proportion, dtype: float64

In [143]:
# Qual a proporção de notas de matemática maiores que 700 do grupo com maior renda familiar (Q006 = Q, acima de R$ 24,2k)?
df['Matematica_maior_700'][df['Q006'] == 'Q'].value_counts(normalize=True).round(2).sort_index(ascending=False)

Matematica_maior_700
True     0.34
False    0.66
Name: proportion, dtype: float64

In [135]:
# Como é a distribuição da nota de matemática de quem teve o pai E a mãe com graduação ou pós graduação?
# Q001: Até que série seu pai, ou o homem responsável por você, estudou?
# Com F = Completou a graduação e G = Completou a pós graduação
df['Matematica_maior_700'][(df['Q001'].isin(['F','G'])) & (df['Q002'].isin(['F','G']))].value_counts(normalize=True).round(2).sort_index(ascending=False)

Matematica_maior_700
True     0.24
False    0.76
Name: proportion, dtype: float64

In [136]:
# Como é a distribuição da nota de matemática de quem teve o pai E a mãe com pós graduação?
df['Matematica_maior_700'][(df['Q001'].isin(['G'])) & (df['Q002'].isin(['G']))].value_counts(normalize=True).round(2).sort_index(ascending=False)

Matematica_maior_700
True     0.26
False    0.74
Name: proportion, dtype: float64

## Conclusões sobre desempenho em matemática

-> A média de matemática foi 542
-> 7% dos que fizeram a prova tiraram mais que 700 
-> Considerando quem tem renda familiar maior que R$ 4,8k, 21% tirou mais que 700
-> Considerando quem tem a maior renda familiar (R$ 24,2k), 34% tirou mais que 700
-> Considerando quem tem pai e mãe com graduação ou pós graduação, 24% tiraram mais que 700
-> Considerando quem tem pai e mãe com pós graduação, 26% tiraram mais que 700